lib/embedded_ansible.rb

```diff
@@ -11,13 +17,114 @@ def self.enabled?
   end
 
   def self.running?
-    # TODO
-    false
+    services.map { |service| LinuxAdmin::Service.new(service).running? }.inject(:&)
+  end
+
+  def self.configure
+    configure_secret_key
+    with_inventory_file do |inventory_file_path|
+      setup_params = {
+        :e => "minimum_var_space=0",
+        :k => "packages,migrations,supervisor",
+        :i => inventory_file_path
+      }
+      AwesomeSpawn.run!(SETUP_SCRIPT, :params => setup_params)
+    end
+    stop
+  end
+
+  def self.start
+    with_inventory_file do |inventory_file_path|
+      setup_params = {
+        :e => "minimum_var_space=0",
+        :k => "packages,migrations",
+        :i => inventory_file_path
+      }
+      AwesomeSpawn.run!(SETUP_SCRIPT, :params => setup_params)
+    end
+  end
+
+  def self.stop
+    services.each { |service| LinuxAdmin::Service.new(service).stop.disable }
+  end
+
+  def self.services
+    AwesomeSpawn.run!("source /etc/sysconfig/ansible-tower; echo $TOWER_SERVICES").output.split
   end
 
   def self.installed?
     path = ENV["APPLIANCE_ANSIBLE_DIRECTORY"] || APPLIANCE_ANSIBLE_DIRECTORY
     Dir.exist?(File.expand_path(path.to_s))
   end
   private_class_method :installed?
+
+  def self.with_inventory_file
+    file = Tempfile.new("miq_inventory")
+    begin
+      file.write(inventory_file_contents)
+      file.close
+      yield(file.path)
+    ensure
+      file.unlink
```

Comment:
I believe the `file.close` is done automatically if you use Tempfile.new with a block.  Are you trying to avoid having the GC do the `unlink` through a finalizer?

In other words, why not:

```ruby
Tempfile.open("miq_inventory") do |file|
  file.write(inventory_file_contents)
  yield(file.path)
end
```